features/include/pcl/features/impl/flare.hpp

```diff
@@ -0,0 +1,318 @@
+    //  float temp;
+    //  Eigen::Vector3f no_centroid = Eigen::Vector3f::Zero();
+    //for(size_t ne = 0; ne < neighbours_indices.size(); ++ne)
+    //{
+    //    no_centroid = (*surface_)[ neighbours_indices[ne] ].getVector3fMap() - centroid;
+
+    //    covariance_matrix (0,0) += no_centroid.x() * no_centroid.x();
+    //    covariance_matrix (1,1) += no_centroid.y() * no_centroid.y();
+    //    covariance_matrix (2,2) = no_centroid.z() * no_centroid.z();
+
+    //    temp =  no_centroid.x() * no_centroid.y();
+    //    covariance_matrix (0,1) += temp;
+    //    covariance_matrix (1,0) += temp;
+    //    temp =  no_centroid.x() * no_centroid.z();
+    //    covariance_matrix (0,2) += temp;
+    //    covariance_matrix (2,0) += temp;
+    //    temp =  no_centroid.y() * no_centroid.z();
+    //    covariance_matrix (1,2) += temp;
+    //    covariance_matrix (2,1) += temp;
+    //  }
+
+    //EIGEN_ALIGN16 Eigen::Vector3f::Scalar eigen_value;
+    //EIGEN_ALIGN16 Eigen::Vector3f eigen_vector;
+    //pcl::eigen33 (covariance_matrix, eigen_value, eigen_vector);
+
+    //fitted_normal.x() = eigen_vector [0];
+    //fitted_normal.y() = eigen_vector [1];
+    //fitted_normal.z() = eigen_vector [2];
+
+    float plane_curvature;
+    normal_estimation_.computePointNormal(*surface_, neighbours_indices, fitted_normal(0), fitted_normal(1), fitted_normal(2), plane_curvature);
+
+    //disambiguate Z axis with normal mean
+    if (!normalDisambiguation<PointNT> (*normals_, neighbours_indices, fitted_normal) )
+    {
+      //all normals in the neighbourood are invalid
+      //setting lrf to NaN
+      lrf.setConstant (std::numeric_limits<float>::quiet_NaN ());
+      return (std::numeric_limits<SignedDistanceT>::max ());
+    }
+  }
+
+  //setting LRF Z axis
+  lrf.row (2).matrix () = fitted_normal;
+
+
+  //find X axis
+
+  //extract support points for Rx radius
+  n_neighbours = sampled_tree_->radiusSearch( (*input_)[index], tangent_radius_, neighbours_indices, neighbours_distances);
+
+  if (n_neighbours < min_neighbors_for_tangent_axis_)
+  {
+    //set X axis as a random axis
+    randomOrthogonalAxis (fitted_normal, x_axis);
+    y_axis = fitted_normal.cross (x_axis);
+
+    return (std::numeric_limits<SignedDistanceT>::max ());
+  }
+
+  //find point with the largest signed distance from tangent plane
+
+  SignedDistanceT shapeScore;
+  SignedDistanceT bestShapeScore = -std::numeric_limits<SignedDistanceT>::max();
+  int bestShapeIndex = -1;
+
+  Eigen::Vector3f best_margin_point;
+
+  float radius2 = tangent_radius_ * tangent_radius_;
+
+  float margin_distance2 = margin_thresh_ * margin_thresh_ * radius2;
+
+
+  Vector3fMapConst feature_point = (*input_)[index].getVector3fMap ();
+
+
+  for (int curr_neigh = 0; curr_neigh < n_neighbours; ++curr_neigh)
+  {
+    const int& curr_neigh_idx = neighbours_indices[curr_neigh];
+    const float& neigh_distance_sqr = neighbours_distances[curr_neigh];
```

Comment:
I'd be the second person ;) This is how you make an alias name to save typing later. Compiler optimizes this out anyway.